packages/expo/src/Notifications/ExponentNotifications.web.ts

```diff
@@ -113,4 +118,92 @@ export default {
   async cancelAllScheduledNotificationsAsync(): Promise<void> {
     this.dismissNotification();
   },
+
+  async getExponentPushTokenAsync(): Promise<string> {
+    if (!Constants.manifest.owner || !Constants.manifest.slug) {
+      throw new CodedError(
+        'ERR_WEB_PUSH_NOTIFICATIONS_MISSING_CONFIG',
+        'You must provide `owner` and `slug` in `app.json` to use push notifications on web. Read more here: https://docs.expo.io/versions/latest/guides/using-vapid/.'
+      );
+    }
+
+    const data = await _subscribeUserToPushAsync();
+    const experienceId = `@${Constants.manifest.owner}/${Constants.manifest.slug}`;
+    const tokenArguments: { [key: string]: string } = {
+      deviceId: Constants.installationId,
+      experienceId: experienceId,
+      // Also uses `experienceId` for `appId` because there's no `appId` for web.
+      appId: experienceId,
+      deviceToken: JSON.stringify(data),
+      type: 'web',
+    };
+
+    // TODO: Use production URL
+    const response = await fetch('http://expo.test/--/api/v2/push/getExpoPushToken', {
+      method: 'POST',
+      body: JSON.stringify(tokenArguments),
+    }).then(response => response.json());
+
+    // TODO: Error handling
+    return response.data.expoPushToken;
+  },
+
+  async getDevicePushTokenAsync(): Promise<{ type: string; data: Object }> {
+    const data = await _subscribeUserToPushAsync();
+    return { type: 'web', data: data };
+  },
 };
+
+async function _subscribeUserToPushAsync(): Promise<Object> {
+  if (!Constants.manifest.notification || !Constants.manifest.notification.vapidPublicKey) {
+    throw new CodedError(
+      'ERR_WEB_PUSH_NOTIFICATIONS_MISSING_CONFIG',
+      'You must provide `notification.vapidPublicKey` in `app.json` to use push notifications on web. Read more here: https://docs.expo.io/versions/latest/guides/using-vapid/.'
+    );
+  }
+  if (!('serviceWorker' in navigator)) {
+    throw new Error(
+      `Current browser does not support Service Worker and hence do not support web push notifications`
+    );
+  }
+
+  const registration = await navigator.serviceWorker.register('/custom-service-worker.js');
```

Comment:
Where does this come from? Is `custom-service-worker.js` the best name for an internal SW

Reply:
I explained about that here: https://github.com/expo/expo-cli/pull/844#issuecomment-512917387. Though I do agree the name is not ideal😅.